rasa/utils/tensorflow/model_data_utils.py

```diff
@@ -434,8 +427,9 @@ def _extract_features(
             dense_features[key].append(value)
 
         # add additional dimensions to attribute mask to get a 3D vector
-        # resulting shape dialogue length x 1 x 1
-        attribute_mask = np.expand_dims(np.expand_dims(attribute_mask, -1), -1)
+        # resulting shape dialogue length x 1
+        # but keep attribute_mask in 3d: (batch x dialogue_len x 1)

```

Comment:
this comment is a bit confusing, `resulting shape` of what?

Reply:
rephrased